src/apps/HeyGPT/HeyGPT/Activation/ProtocolActivationHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Diagnostics;
+using Windows.ApplicationModel.Activation;
+using Microsoft.Windows.AppLifecycle;
+using AndGPT.Core.Contracts.Services;
+using AndGPT.Core.Events;
+
+namespace HeyGPT.App.Activation;
+
+public sealed class ProtocolActivationHandler : ActivationHandler<AppActivationArguments>
```

Comment:
The protocol handler handles any URI activation.   We have registered both the hey:// and gpt:// in the package manifest.   if either of these URI schemes are run in the windows run or by clicking a link this application will be activated.    This will parse the URIs and then raise an event aggregator event.